Row.c

```diff
@@ -442,6 +442,10 @@ void Row_printLeftAlignedField(RichString* str, int attr, const char* content, u
 }
 
 int Row_printPercentage(float val, char* buffer, size_t n, uint8_t width, int* attr) {
+   assert(width >= 4 && width < n && "Invalid width in Row_printPercentage()");
+   // truncate in favour of abort in xSnprintf()
+   width = (uint8_t)CLAMP(width, 4, n - 1);
```

Comment:
@cgzones The maximum width should be `n - 2`. Because the format string of `xSnprintf` below has a trailing space.
Or alternatively just use `snprintf` instead of `xSnprintf`?

Reply:
I'd prefer `n-2` for the fix here. The `xSnprintf` is there to catch such issues.

Reply:
And what if `n` is too small? `n - 2 < 4` for the `CLAMP` macro

Reply:
I noticed there to be an assert (`n > width`) after that clamping to be still missing. Will push some additional fix for that part later; hopefully resolving this entirely.